bokehjs/src/coffee/models/markers/all_markers.coffee

```diff
@@ -0,0 +1,204 @@
+M = require "./marker"
+
+SQ3 = Math.sqrt(3)
+
+_one_x =  (ctx, r) ->
+  ctx.moveTo(-r,  r)
+  ctx.lineTo( r, -r)
+  ctx.moveTo(-r, -r)
+  ctx.lineTo( r,  r)
+
+_one_cross = (ctx, r) ->
+  ctx.moveTo( 0,  r)
+  ctx.lineTo( 0, -r)
+  ctx.moveTo(-r,  0)
+  ctx.lineTo( r,  0)
+
+_one_tri = (ctx, r) ->
+  h = r * SQ3
+  a = h/3
+
+  # TODO (bev) use viewstate to take y-axis inversion into account
+  ctx.moveTo(-r, a)
+  ctx.lineTo(r, a)
+  ctx.lineTo(0, a-h)
+  ctx.closePath()
+
+_asterisk = (ctx, i, sx, sy, r, vl, vf) ->
+  r2 = r*0.65
+
+  _one_cross(ctx, r)
+  _one_x(ctx, r2)
+
+  if vl.doit
+    vl.set_vectorize(ctx, i)
+    ctx.stroke()
+
+  return
+
+_circle_cross = (ctx, i, sx, sy, r, vl, vf)  ->
+  ctx.arc(0, 0, r, 0, 2*Math.PI, false)
+
+  if vf.doit
+    vf.set_vectorize(ctx, i)
+    ctx.fill()
+
+  if vl.doit
+    vl.set_vectorize(ctx, i)
+    _one_cross(ctx, r)
+    ctx.stroke()
+
+  return
+
+_circle_x = (ctx, i, sx, sy, r, vl, vf) ->
+  ctx.arc(0, 0, r, 0, 2*Math.PI, false)
+
+  if vf.doit
+    vf.set_vectorize(ctx, i)
+    ctx.fill()
+
+  if vl.doit
+    vl.set_vectorize(ctx, i)
+    _one_x(ctx, r)
+    ctx.stroke()
+
+  return
+
+_cross = (ctx, i, sx, sy, r, vl, vf) ->
+  _one_cross(ctx, r)
+
+  if vl.doit
+    vl.set_vectorize(ctx, i)
+    ctx.stroke()
+
+  return
+
+_diamond = (ctx, i, sx, sy, r, vl, vf) ->
+  ctx.moveTo(0, r)
+  ctx.lineTo(r/1.5, 0)
+  ctx.lineTo(0, -r)
+  ctx.lineTo(-r/1.5, 0)
+  ctx.closePath()
+
+  if vf.doit
+    vf.set_vectorize(ctx, i)
+    ctx.fill()
+
+  if vl.doit
+    vl.set_vectorize(ctx, i)
+    ctx.stroke()
+
+  return
+
+_diamond_cross = (ctx, i, sx, sy, r, vl, vf) ->
+  ctx.moveTo(0, r)
+  ctx.lineTo(r/1.5, 0)
+  ctx.lineTo(0, -r)
+  ctx.lineTo(-r/1.5, 0)
+  ctx.closePath()
+
+  if vf.doit
+    vf.set_vectorize(ctx, i)
+    ctx.fill()
+
+  if vl.doit
+    vl.set_vectorize(ctx, i)
+    _one_cross(ctx, r)
+    ctx.stroke()
+
+  return
+
+_inverted_triangle = (ctx, i, sx, sy, r, vl, vf) ->
+  ctx.rotate(Math.PI)
+  _one_tri(ctx, r)
+  ctx.rotate(-Math.PI)
+
+  if vf.doit
+    vf.set_vectorize(ctx, i)
+    ctx.fill()
+
+  if vl.doit
+    vl.set_vectorize(ctx, i)
+    ctx.stroke()
+
+  return
+
+_square = (ctx, i, sx, sy, r, vl, vf) ->
+  size = 2*r
+  ctx.rect(-r, -r, size, size)
+
+  if vf.doit
+    vf.set_vectorize(ctx, i)
+    ctx.fill()
+
+  if vl.doit
+    vl.set_vectorize(ctx, i)
+    ctx.stroke()
+
+  return
+
+_square_cross = (ctx, i, sx, sy, r, vl, vf) ->
+  size = 2*r
+  ctx.rect(-r, -r, size, size)
+
+  if vf.doit
+    vf.set_vectorize(ctx, i)
+    ctx.fill()
+
+  if vl.doit
+    vl.set_vectorize(ctx, i)
+    _one_cross(ctx, r)
+    ctx.stroke()
+
+  return
+
+_square_x = (ctx, i, sx, sy, r, vl, vf) ->
+  size = 2*r
+  ctx.rect(-r, -r, size, size)
+
+  if vf.doit
```

Comment:
Also FYI SourceCodePro (free font) was designed to disambiguate 1/l and 0/O etc 

> On Aug 23, 2016, at 07:29, Bryan Van de Ven <bryanv@continuum.io> wrote:
> 
> If the minifier will shorten them now that they are passed in as parameters then sure, agree. 
> 
>> On Aug 23, 2016, at 03:35, Mateusz Paprocki <notifications@github.com> wrote:
>> 
>> In bokehjs/src/coffee/models/markers/all_markers.coffee:
>> 
>> > +  if vf.doit
>> > +    vf.set_vectorize(ctx, i)
>> > +    ctx.fill()
>> > +
>> > +  if vl.doit
>> > +    vl.set_vectorize(ctx, i)
>> > +    _one_cross(ctx, r)
>> > +    ctx.stroke()
>> > +
>> > +  return
>> > +
>> > +_square_x = (ctx, i, sx, sy, r, vl, vf) ->
>> > +  size = 2*r
>> > +  ctx.rect(-r, -r, size, size)
>> > +
>> > +  if vf.doit
>> Proper naming should have been kept. It took me a while to figure out that vl stands for and that it's not actually v1 (stupid fonts).
>> 
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub, or mute the thread.
>>